docusaurus/docs/ui-extensions/create-an-extension-point.md

```diff
@@ -101,6 +341,8 @@ Component type extensions are simple React components, which gives much more fre
 import { getPluginComponentExtensions } from '@grafana/runtime';
 
 export const Toolbar = () => {
+  // This only returns type="component" extensions
+  // Heads up! We don't specify a context object below, we pass in the contextual information as a prop to the component later.
```

Comment:
Could you help me understand why we're doing it this way and what we want to ensure the reader takes away having read this comment?

Reply:
We did it this way because we thought that it's more natural for React components to receive input through component props, and this pattern can save an extra step for the plugin dev, they don't need to proxy the context object that they would receive in the extension registration callback to the  extension component, they can just rely on what the extension point passes down to them. The extension point can even pass down multiple props, if they thing it's more meaningful / semantical.

> what we want to ensure the reader takes away having read this comment?

I basically wanted to raise their attention that passing down contextual information to components is different than for links. 

Probably there is some discrepancy here that we should chat through though, we should try making this as easy to understand as possible.